tests/Unit/NumericalAlgorithms/Integration/Test_GslQuadAdaptive.cpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cmath>
+#include <cstddef>
+
+#include "Framework/TestHelpers.hpp"
+#include "NumericalAlgorithms/Integration/GslQuadAdaptive.hpp"
+
+namespace {
+/// [integrated_function]
+double gaussian(double x, double mean, double factor) {
+  return 2. * factor / sqrt(M_PI) * exp(-square(x - mean));
+}
+
+/// [integrated_function]
+double integrable_singularity(double x, double factor) {
+  return factor * cos(sqrt(abs(x))) / sqrt(abs(x));
+}
+
+SPECTRE_TEST_CASE("Unit.Numerical.Integration.GslQuadAdaptive",
+                  "[Unit][NumericalAlgorithms]") {
+  {
+    INFO("StandardGaussKronrod");
+    // Construct the integration and give an example
+    /// [integration_example]
+    integration::GslQuadAdaptive<
+        integration::GslIntegralType::StandardGaussKronrod>
+        integration{20};
+    const double mean = 5.;
+    const double factor = 2.0;
+    const double lower_boundary = -4.;
+    const double upper_boundary = 10.;
+    const auto result = integration(
+        [&mean, &factor](const double x) { return gaussian(x, mean, factor); },
+        lower_boundary, upper_boundary, 1.e-10, 4);
+    /// [integration_example]
+    CHECK(result == approx(factor * erf(upper_boundary - mean) -
```

Comment:
why are you checking that the results are more accurate than 1.e-10?

Reply:
I have since changed the accuracy to 1.e-11 to match the accuracy of approx(). I hope that makes more sense.

Reply:
I think it would be better to explicitly set the precision of the `approx` so it matches the error bound you are requesting from GSL. You can add this to be beginning of the test (before line 27)

```cpp
    const double absolute_tolerance = 1.e-10;
    Approx custom_approx = Approx::custom().epsilon(absolute_tolerance);
```

And then use `absolute_tolerance` when calling integration functions and `custom_approx` when checking results.